src/rsz/src/RepairSetup.cc

```diff
@@ -579,16 +603,142 @@ bool RepairSetup::swapPins(PathRef* drvr_path,
 
 // Remove driver if
 // 1) it is a buffer without attributes like dont-touch
-// 2) it improves setup slack
-// 3) it doesn't create new max fanout violations
-// 4) it doesn't create new max cap violations
-// 5) it doesn't create new max slew violations
+// 2) it doesn't create new max fanout violations
+// 3) it doesn't create new max cap violations
+// 4) it doesn't worsen slack
 bool RepairSetup::removeDrvr(PathRef* drvr_path,
                              LibertyCell* drvr_cell,
                              int drvr_index,
                              PathExpanded* expanded)
 {
-  /* TODO: remove driver only if slack improves and no new violations */
+  // TODO:
+  // 1. add setup slack check (based on Elmore delay?)
+  // 2. check dont-touch on nets
+  // 3. add slew check
+  if (drvr_cell && drvr_cell->isBuffer()) {
+    Pin* drvr_pin = drvr_path->pin(this);
+    Instance* drvr = network_->instance(drvr_pin);
+
+    // Don't remove buffers from previous sizing, pin swapping, rebuffering, or
+    // cloning because such removal may lead to an inifinte loop or long runtime
+    if (resizer_->all_sized_inst_set_.count(drvr)) {
+      debugPrint(logger_,
+                 RSZ,
+                 "repair_setup",
+                 4,
+                 "buffer {} is not removed because it has been resized",
+                 db_network_->name(drvr));
```

Comment:
This should probably be a single message with the reason as a variable. Will reduce lines of code. e.g.

```
std::string reason;
if (resizer_->all_sized_inst_set_.count(drvr)) {
  reason = "it has been resized.";
else if (resizer_->all_swapped_pin_inst_set_.count(drvr)) {
  reason = "its pins have been swapped.";
}
// ...

"buffer {} is not removed because {}",
db_network_->name(drvr)),
reason
```

Reply:
Good idea.  Fixed.